### ISSUE TITLE
Catch `variable is []`

### ISSUE THREAD

**Issue by francorbacho** [astral-sh/ruff#8589]
Make [F632](https://docs.astral.sh/ruff/rules/is-literal/) catch the following trap:

```python
variable = []
if variable is []:
  print("never runs")
```

Tested in Arch Linux:

```bash
$ pacman -Q ruff 
ruff 0.1.5-1
```

**Comment by charliermarsh** [comment]
It seems reasonable to me.

**Comment by francorbacho** [comment]
Although maybe it should be a different rule if we want to also test for `variable is [0, 1, 2]`.

**Comment by charliermarsh** [comment]
Seems ok to fold it into the same rule for me, since that will also always be false, right?

**Comment by jesse1412** [comment]
Could this be expanded to cover any of the mutable initialisers? E.g set, dict, etc?

**Comment by charliermarsh** [comment]
I believe this was closed by #8607 :) Thanks @jesse1412!
